libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java

```diff
@@ -534,8 +532,9 @@ private static byte[] decryptWithPrivateKey(PrivateKey privateKey, String data,
         return null;
     }
 
-    private static byte[] generateInitVector() throws NoSuchAlgorithmException {
-        final SecureRandom random = SecureRandom.getInstance(SHA1PRNG);
+    private static byte[] generateInitVector() {
+        // Create the system recommended secure random number generator provider algorithm.
+        final SecureRandom random = new SecureRandom();
```

Comment:
The documentation I referenced in the description indicates it's preferred to use this constructor which allows `SecureRandom` to choose the recommended available algorithm.